File: api/internal/common/schemas/training_session.py
from typing import List, Optional
from fastapi import Query
from pydantic import BaseModel, Field

from internal.common.enums.training_session import TrainingSessionType
from internal.common.schemas.common import ListResponse, PagingRequest
from internal.common.schemas.teams import TeamResponse
from internal.common.schemas.user import UserInfo
from internal.common.types import ID


class TrainingSessionInfo(BaseModel):
    name: str = Field(..., description="Training session name")
    coach_id: Optional[ID] = Field(None, description="Coach ID")
    team_id: Optional[ID] = Field(None, description="Team ID")
    start_date: Optional[int] = Field(..., description="Start date")
    session_type: Optional[TrainingSessionType] = Field(
        ..., description="Training Session Type")


class CreateTrainingSessionRequest(TrainingSessionInfo):
    ...


class UpdateTrainingSessionRequest(TrainingSessionInfo):
    ...


class TrainingSessionResponse(TrainingSessionInfo):
    id: ID = Field(..., description="ID")
    coach: Optional[UserInfo] = Field(None, description="Coach info")
    team: Optional[TeamResponse] = Field(None, description="Team info")
    created_at: int = Field(..., description="Team created at")
    updated_at: int = Field(..., description="Team updated at")


class ListTrainingSessionRequest(PagingRequest):
    type: Optional[TrainingSessionType] = Field(
        Query(None, description="training session type"))


class ListTrainingSessionResponse(ListResponse):
    training_sessions: List[TrainingSessionResponse] = Field(
        ..., description="List training sessions")

File: web/lib/schemas/trainingsession.ts
import { z } from "zod";
import { userSchema } from "./user";
import { teamSchema } from "./teams";

export const sessionTypeEnum = z.enum(["Training", "Match", "Evaluation"]);

export type SessionType = z.TypeOf<typeof sessionTypeEnum>;

export const trainingSessionSchema = z.object({
  id: z.string(),
  name: z.string(),
  coach_id: z.string(),
  team_id: z.string(),
  start_date: z.number(),
  session_type: sessionTypeEnum,
  coach: userSchema,
  team: teamSchema,
  created_at: z.number(),
  updated_at: z.number(),
});

export type TrainingSession = z.TypeOf<typeof trainingSessionSchema>;

export const trainingSessionCreateSchema = trainingSessionSchema.pick({
  name: true,
  coach_id: true,
  team_id: true,
  start_date: true,
  session_type: true,
});

export type TrainingSessionCreate = z.TypeOf<
  typeof trainingSessionCreateSchema
>;

export const trainingSessionUpdateSchema = trainingSessionSchema.pick({
  id: true,
  name: true,
  coach_id: true,
  team_id: true,
  start_date: true,
  session_type: true,
});

export type TrainingSessionUpdate = z.TypeOf<
  typeof trainingSessionUpdateSchema
>;
